### ISSUE TITLE
Running example results in lower evaluation scores than stated on GitHub readme

### ISSUE THREAD

**Issue by matthias-samwald** [facebookresearch/PyTorch-BigGraph#82]
I followed the basic instructions in the GitHub Readme, however, the results I get are lower than described in the Readme:

The last few lines of the output when running the eval script: 
```
...
019-07-11 11:16:28  WARNING: Adding uniform negatives makes no sense when already using all negatives
2019-07-11 11:16:28  ( 0 , 0 ): Processed 59071 edges in 5.6 s (0.011M/sec); load time: 0.0015 s
2019-07-11 11:16:28  Stats for edge path 1 / 1, bucket ( 0 , 0 ): pos_rank:  209.651 , mrr:  0.245 , r1:  0.136133 , r10:  0.492543 , r50:  0.720726 , auc:  0.989885 , count:  59071
2019-07-11 11:16:28
2019-07-11 11:16:28  Stats for edge path 1 / 1: pos_rank:  209.651 , mrr:  0.245 , r1:  0.136133 , r10:  0.492543 , r50:  0.720726 , auc:  0.989885 , count:  59071
2019-07-11 11:16:28
2019-07-11 11:16:28
2019-07-11 11:16:28  Stats: pos_rank:  209.651 , mrr:  0.245 , r1:  0.136133 , r10:  0.492543 , r50:  0.720726 , auc:  0.989885 , count:  59071
```

The Readme states that the last line should be:
`Stats: pos_rank:  65.4821 , mrr:  0.789921 , r1:  0.738501 , r10:  0.876894 , r50:  0.92647 , auc:  0.989868 , count:  59071`

Are there any explanations?

I think this issue was already raised in https://github.com/facebookresearch/PyTorch-BigGraph/issues/18 (but was misunderstood as pointing out a discrepancy with the paper)

**Comment by adamlerer** [comment]
Can you post the exact command you ran that led to these results? As mentioned in #18, the fb15k results in the literature use a different eval metric (filtered MRR), the best way to run fb15k with all the right settings is using the `examples/fb15k.py` script in the repo, which does both train and eval.

**Comment by jiajunshen** [comment]
@matthias-samwald You are right: By following the command in README, you should get the result you are getting right now. The result stated in ReadMe is achieved by `examples/fb15k.py`. We will update the README file.

**Comment by matthias-samwald** [comment]
I ran the sequence of commands listed in the ReadMe, i.e., https://github.com/facebookresearch/PyTorch-BigGraph

I'm only trying to replicate the ReadMe, not the literature results.

**Comment by adamlerer** [comment]
Ah, I see the confusion. The README walks through the equivalent steps that you could take in order to replicate `examples/fb15k.py`, however, you can't match the eval via the command-line tool `torchbiggraph_eval` because it doesn't support filtered eval. There's a paragraph that explains this in the README:

```
The evaluation performed by the torchbiggraph_example_fb15k command differs from the above torchbiggraph_eval command, in order to match the literature. It calculates the ranks of the edges in the evaluation set by comparing them against all other edges except the ones that are true positives in any of the training, validation or test set. This setup, called filtered MRR, is only used to evaluate small graphs because it scales very poorly.
```

but you're right that it's not clear at all, and the README makes it seem like the MRR of 0.789 is from running `torchbiggraph_eval`. I will fix the README to make it clear.

**Comment by matthias-samwald** [comment]
Thanks! Indeed it was very confusing.
